Sources/PostgresNIO/Connection/PostgresConnection.swift

```diff
@@ -78,9 +78,15 @@ public final class PostgresConnection {
             /// - Default: 5432
             public var port: Int
 
-            public init(host: String, port: Int = 5432) {
+            /// Specifies a timeout to apply to a connection attempt.
+            ///
+            /// - Default: 10 seconds
+            public var connectTimeout: TimeAmount
+
+            public init(host: String, port: Int = 5432, connectTimeout: TimeAmount = .seconds(10)) {
                 self.host = host
                 self.port = port
+                self.connectTimeout = connectTimeout
```

Comment:
```suggestion
            public init(host: String, port: Int = 5432) {
                self.host = host
                self.port = port
                self.connectTimeout = .seconds(10)
```
We sadly can not add another parameter to the initializer and not be source breaking. To work around this all properties are `vars` by default. This way, if users want to override the value, they can just use the property after the init.

Reply:
No problem, but I have to admit that I don't understand why this is source breaking... 🙂 Shouldn't a new parameter with a default value at the end of the parameter list compile everywhere without problems?

Reply:
Yes, this API breaking change is very hard to see, but it occurs in this case:

Existing API:
```swift
struct Foo {
    init(bar: String) {

    }

    static func test() {
        let make = Foo.init(bar:)
        let foo = make("hello")
        // ...
    }
}
```

Adding a defaulted value breaks this code however:
```swift
struct Foo {
    init(bar: String, test: String = "foo") {

    }

    static func test() {
        let make = Foo.init(bar:) // <--- ❌ compiler error
        let foo = make("hello")
        // ...
    }
}
```